app/src/main/java/org/schabi/newpipe/player/MainVideoPlayer.java

```diff
@@ -394,6 +400,11 @@ protected void setShuffleButton(final ImageButton shuffleButton, final boolean s
         shuffleButton.setImageAlpha(shuffleAlpha);
     }
 
+    protected void setMuteButton(final ImageButton muteButton, final boolean isMuted) {
+        muteButton.setColorFilter(ContextCompat.getColor(getApplicationContext(), isMuted ? R.color.white : R.color.gray));
```

Comment:
What if the theme is white theme? The active button color should be black in that case. I'd suggest adding two attributes in `attrs.xml` named something along the lines of `button_active` and `button_inactive` and giving them different color values in the `themes.xml` file, so that they are themed.

Reply:
White theme does not affect background color for Main video player. Background is always black for this player.

Reply:
Oh you are right, nevermind :man_facepalming: 
Then there should be different code for every player, at this point

Reply:
For MainVideoPlayer icon change happens in setMuteButton (place you've mentioned above). 
For PopUp and Background, icon change happens for queue in ServicePlayerActivity in onMaybeMuteChanged. I did it similarly to other buttons, (e.g onPlayModeChanged also in ServicePlayerActivity)